src/main/Numbers.java

```diff
@@ -0,0 +1,41 @@
+package main;
+
+import java.util.Collections;
+import java.util.List;
+
+public class Numbers<T extends Number> {
+    private List<T> numbersList;
+
+    public Numbers(List<T> numbersList) {
+        this.numbersList = numbersList;
+    }
+
+    public int countEvenNumbers() {
+        int counter = 0;
+        for (T number : numbersList) {
+            if (number.doubleValue() % 2 == 0) {
+                counter++;
+            }
+        }
+        return counter;
+    }
+
+    /**
+     * This method sets list 1's element with an index corresponding to index parameter
+     * to a value of an element of list 2 with the same index preventing the reverse operation.
+     * */
+    public void setValue(List<T> list1, List<T> list2, int index) {
+        list2 = Collections.unmodifiableList(list2);
+        list1.set(index, list2.get(index));
```

Comment:
Nice one but still, the code `list2.set(index, list1.get(index));` compiles and fails only in runtime. I believe that this error can be found at compile-time. Take a look at wildcard upper and lower bounds.